pkgs/tools/misc/asdf-vm/default.nix

```
@@ -0,0 +1,55 @@
+{ stdenv, lib, fetchFromGitHub, makeWrapper, installShellFiles, bash, coreutils, curl, git }:
+
+stdenv.mkDerivation rec {
+  pname = "asdf-vm";
+  version = "0.8.1";
+
+  src = fetchFromGitHub {
+    owner = "asdf-vm";
+    repo = "asdf";
+    rev = "v${version}";
+    sha256 = "07lh5k1krzm7fbkv0jlwbzz5ycn2jg7s12dnqwmy82qqic0sl6fl";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    installShellFiles
+  ];
+
+  buildInputs = [
+    bash
+    curl
+    git
+  ];
+
+  installPhase = ''
+    mkdir -p $out/share/asdf-vm
+    cp -r . $out/share/asdf-vm
+
+    mkdir -p $out/etc/profile.d
+    echo "source $out/share/asdf-vm/asdf.sh" > $out/etc/profile.d/asdf.sh
+
+    mkdir -p $out/bin
+    # asdf-vm create "shim" file like this:
+    #
+    #    exec $ASDF_DIR/bin/asdf exec "node" "$@"
+    #
+    # So we shouldn't set ASDF_DIR like $out/share/asdf-vm, because $out
+    # always change
+    makeWrapper $out/share/asdf-vm/bin/asdf $out/bin/asdf \
+      --run "export ASDF_DIR=\"\$HOME/.nix-profile/share/asdf-vm\""
```

Hi @SuperSandro2000, do you think I resolve this problem in the right way?

---

I am not sure. ``~/.nix-profile/share/asdf-vm`` is read-only on my system and is only created when installing the package, not when using it in nix-shell. We should probably hardcode this to the $out path.

```suggestion
      --set ASDF_DIR $out/share/asdf-vm
```

---

I see, I just realized the code `echo "source $out/share/asdf-vm/asdf.sh" > $out/etc/profile.d/asdf.sh` will not work in nix-shell also

I tried a new idea (you can see my new temporary commit), but it's only working when upgraded asdf-vm and execute `asdf` command at first. If I use asdf-vm 0.7.8 installed nodejs, then upgrade asdf-vm, then execute `nix-collect-garbage`, then execute `node` immediately, shell will throw an error like `/nix/store/gmkxfwl44lxfr9li8bk9f0qdr5xz8vpv-asdf-vm-0.7.8/share/asdf-vm/bin/asdf: No such file or directory`

And also, I realized that `echo "source $out/share/asdf-vm/asdf.sh" > $out/etc/profile.d/asdf.sh` not working in nix-shell, but it is very important for asdf-vm 😂 

Do you have any good idea to resolve those problems?